File: backend/core/src/modules/writing-goals/writing-goals.controller.ts
import { Controller, UseGuards } from '@nestjs/common';
import { ProjectScopedCrudController } from '../../common/access/project-scoped-crud.controller';
import { WritingGoalsService } from './writing-goals.service';
import { WritingGoal } from '../../entities';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';

@UseGuards(JwtAuthGuard)
@Controller('writing-goals')
export class WritingGoalsController extends ProjectScopedCrudController<WritingGoal> {
  constructor(service: WritingGoalsService) {
    super(service);
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/WritingGoalsController.java
package store.pjcloud.cowrite.core.controller;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import store.pjcloud.cowrite.core.common.ProjectScopedCrudController;
import store.pjcloud.cowrite.core.entity.WritingGoal;
import store.pjcloud.cowrite.core.service.WritingGoalsService;

@RestController
@RequestMapping("/writing-goals")
public class WritingGoalsController extends ProjectScopedCrudController<WritingGoal> {
    public WritingGoalsController(WritingGoalsService service) {
        super(service);
    }
}
